libraries/networking/src/ExternalResource.cpp

```diff
@@ -49,13 +49,32 @@ QUrl ExternalResource::getQUrl(Bucket bucket, const QUrl& path) {
         base = _bucketBases[bucket];
     }
 
-    QUrl merged = base.resolved(path).adjusted(QUrl::NormalizePathSegments);
+    QUrl filteredPath = path;
+    QString pathQString = filteredPath.path();
+
+    // De-duplicate URL separators, since S3 doesn't actually have subdirectories, and treats a '/' as a part
+    // of the filename. This means that "/dir/file.txt" and "/dir//file.txt" are not equivalent on S3.
+    while(pathQString.contains("//")) {
+        pathQString = pathQString.replace("//", "/");
+    }
+
+    // If a path starts with a / it would have the effect of overriding any path specified in the bucket.
+    // We remove it, ensuring that getQUrl(Bucket.Assets, "/file.txt") and getQUrl(Bucket.Assets, "file.txt")
+    // are equivalent.
+    if (pathQString.startsWith("/")) {
+        pathQString.remove(0,1);
+    }
+
+    filteredPath.setPath(pathQString);
+
+    QUrl merged = base.resolved(filteredPath).adjusted(QUrl::NormalizePathSegments);
```

Comment:
```suggestion
    QString filteredPath = path.toString();

    // De-duplicate URL separators, since S3 doesn't actually have subdirectories, and treats a '/' as a part
    // of the filename. This means that "/dir/file.txt" and "/dir//file.txt" are not equivalent on S3.
    filteredPath = filteredPath.replace(QRegExp("(//+)"), "/");

    // If a path starts with a / it would have the effect of overriding any path specified in the bucket.
    // We remove it, ensuring that getQUrl(Bucket.Assets, "/file.txt") and getQUrl(Bucket.Assets, "file.txt")
    // are equivalent.
    if (filteredPath.startsWith("/")) {
        filteredPath.remove(0, 1);
    }

    QUrl merged = base.resolved(QUrl(filteredPath)).adjusted(QUrl::NormalizePathSegments);
```
The use of `,path()` and `.setPath()` causes unexpected behaviour for paths starting with `//`. For example, `//alan/dev/tablet-with-home-button.fbx` ends up with the merged URL containing just `dev/tablet-with-home-button.fbx` because the `//alan` part is treated as a domain.